src/Controller/Api/Carrinho/DescartarCarrinhoController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Controller\Api\Carrinho;
+
+use App\Entity\StatusEnum;
+use App\Repository\CarrinhoRepository;
+use App\Repository\ClienteRepository;
+use App\Service\DescartarCarrinhoService;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Routing\Annotation\Route;
+
+class DescartarCarrinhoController extends AbstractController
+{
+  #[Route('/api/carrinho/descartar/{id}', name: 'decartar_carrinho')]
+  public function index(
+    CarrinhoRepository $carrinhoRepository,
+    DescartarCarrinhoService $descartarCarrinhoService,
+    int $id,
+  ): JsonResponse {
+    $id = $carrinhoRepository->findOneBy(
+      [
+        'cliente' => [
+          'id' => $id
+        ],
+        'status' => StatusEnum::aberto
+      ]
+    )->getId();
+    try {
+      $descartarCarrinhoService->execute($id);
+      return new JsonResponse(['response' => 'Carrinho descartado com sucesso'], 200, [], false);
+    } catch (\Throwable $e) {
+      return $e instanceof \Exception
+        ? new JsonResponse(['response' => $e->getMessage()], 200)
+        : new JsonResponse(['response' => $e->getMessage()], 500);
+    }
+  }
```

Comment:
- ja tem o ID do carrinho então não precisa buscar novamente...
- não precisa olhar a instancia de exception por só vai cair no catch se der erro
- oque é default não precisa passar novamente...

```suggestion
    try {
      $descartarCarrinhoService->execute($id);
      return new JsonResponse(['response' => 'Carrinho descartado com sucesso']);
    } catch (\Throwable $e) {
      return new JsonResponse(['response' => $e->getMessage()], 500);
    }
  }
```

Reply:
esse id que ele ta recebendo é o id do cliente, ta passando la na tela de venda

Reply:
Então muda o nome da varivavel para clienteId para não confundir com o id do Carrinho

Reply:
beleza